localMatchApp/test/logintest.cpp

```diff
@@ -0,0 +1,83 @@
+#include <gmock/gmock-matchers.h>
+#include <gtest/gtest.h>
+#include <iostream>
+#include <string>
+#include "Login.h"
+using namespace testing;
+
+TEST ( LoginTest, TestPasswordGoodCases ) {
+  Login objectOfTest;
+
+  std::string pass1 = "Robert0";
+  std::string pass2 = "R0i";
+  std::string pass3 = "4123412341234123412RR12341234123r";
+  std::string pass4 = "EWRTWERGDFSGSD3DFGDFGSDFr";
+  std::string pass5 = "wertwertwerw4wertwerRtwert";
+  std::string pass6 = "Il0";
+
+  EXPECT_TRUE ( objectOfTest.hasCorrectFormat ( pass1 ) );
+  EXPECT_TRUE ( objectOfTest.hasCorrectFormat ( pass2 ) );
+  EXPECT_TRUE ( objectOfTest.hasCorrectFormat ( pass3 ) );
+  EXPECT_TRUE ( objectOfTest.hasCorrectFormat ( pass4 ) );
+  EXPECT_TRUE ( objectOfTest.hasCorrectFormat ( pass5 ) );
+  EXPECT_TRUE ( objectOfTest.hasCorrectFormat ( pass6 ) );
+}
+
+TEST ( LoginTest, TestPasswordBadCases ) {
```

Comment:
provide a test description of what you are trying to verify

Reply:
Done.